corehq/apps/case_search/models.py

```diff
@@ -161,7 +161,10 @@ def replace_custom_query_variables(query_addition, criteria, ignore_patterns):
             for removal_regex in remove_char_regexs:
                 to_remove = re.escape(removal_regex.regex)
                 value = re.sub(to_remove, '', value)
-        query_addition = re.sub(key, value, query_addition)
+        to_add = re.escape(value)
+        query_addition = re.sub(key, to_add, query_addition)
+
+    query_addition = re.sub(re.escape('\\'), re.escape('\\\\'), query_addition)
```

Comment:
It would be simpler to not use a regular expression here.
```
query_addition = query_addition.replace('\\', '\\\\')
```

Reply:
updated thanks